platform/ios/src/MGLMapView.mm

```diff
@@ -967,40 +981,44 @@ - (void)layoutSubviews
 /// Updates `contentInset` to reflect the current window geometry.
 - (void)adjustContentInset
 {
-    // We could crawl all the way up the responder chain using
-    // -viewControllerForLayoutGuides, but an intervening view means that any
-    // manual contentInset should not be overridden; something other than the
-    // top and bottom bars may be influencing the manual inset.
-    UIViewController *viewController;
-    if ([self.nextResponder isKindOfClass:[UIViewController class]])
-    {
-        // This map view is the content view of a view controller.
-        viewController = (UIViewController *)self.nextResponder;
-    }
-    else if ([self.superview.nextResponder isKindOfClass:[UIViewController class]])
+    UIEdgeInsets adjustedContentInsets = UIEdgeInsetsZero;
+    
+    if (@available(iOS 11.0, *))
     {
-        // This map view is an immediate child of a view controller’s content view.
-        viewController = (UIViewController *)self.superview.nextResponder;
-    }
+        adjustedContentInsets = self.safeAreaInsets;
+        
+    } else {
+        // We could crawl all the way up the responder chain using
+        // -viewControllerForLayoutGuides, but an intervening view means that any
+        // manual contentInset should not be overridden; something other than the
+        // top and bottom bars may be influencing the manual inset.
+        UIViewController *viewController;
+        if ([self.nextResponder isKindOfClass:[UIViewController class]])
+        {
+            // This map view is the content view of a view controller.
+            viewController = (UIViewController *)self.nextResponder;
+        }
+        else if ([self.superview.nextResponder isKindOfClass:[UIViewController class]])
+        {
+            // This map view is an immediate child of a view controller’s content view.
+            viewController = (UIViewController *)self.superview.nextResponder;
+        }
+        
+        adjustedContentInsets.top = viewController.topLayoutGuide.length;
+        CGFloat bottomPoint = CGRectGetMaxY(viewController.view.bounds) -
+                                (CGRectGetMaxY(viewController.view.bounds)
+                                - viewController.bottomLayoutGuide.length);
+        adjustedContentInsets.bottom = bottomPoint;
 
-    if ( ! viewController.automaticallyAdjustsScrollViewInsets)
```

Comment:
This is breaking change for people using automaticallyAdjustsScrollViewInsets.
We are not replacing it by contentInsetAdjustmentBehavior that documentation suggests:
 
@property(nonatomic,assign) BOOL automaticallyAdjustsScrollViewInsets API_DEPRECATED("Use UIScrollView's contentInsetAdjustmentBehavior instead", ios(7.0,11.0),tvos(7.0,11.0)); // Defaults to YES

Just a comment to reconsider if releasing this in minor release.

Reply:
Indeed — we need to keep supporting `automaticallyAdjustsScrollViewInsets`, especially for apps that have a minimum deployment version lower than iOS 11.

Reply:
Any ideas on how to signal that we will not use this variable?

Reply:
It’s a semver major change if we stop, so that’s not really an option right now.

Reply:
I added the warning.

Reply:
Has @friedbunny's server concern here been addressed?

Reply:
@julianrex yes we are supporting both properties and logging a warning.